### DISCUSSION TITLE
new fabric.Canvas

### DISCUSSION THREAD

**Discussion by ye001226** [fabricjs/fabric.js#9720] [Q&A]
为什么在初始化后会多一个canvas-container的div和一个canvas？
为什么不能在我原有的canvas上完成工作？

**Comment by ye001226** [comment]
version 5.3.0

**Reply by dark-passages** [reply]
That does not answer the question!
Not even sure if you mean "There will be no need for the additional stuf in version 5.3." 
Could you be more vague?

**Reply by dark-passages** [reply]
O wait, i'm sure you don't mean that... in 5.3.1 the additional stuff is still there, but why?

**Comment by asturur** [comment]
There is the additional stuff because of retina scaling and the interactive layer.
In theory with a bit of work those are not exactly necessary, you can demand css scaling to a container made by you but as of now the upper canvas, required for text editing, brushing and selection requires a container to be aligned.

It is something that was done in around 2010 and never touched, and was fine at the time.
Is fine also today but different ways to initialize a canvas can also be planned and made.
